core/src/main/scala/akka/kafka/scaladsl/Consumer.scala

```diff
@@ -128,11 +128,22 @@ object Consumer {
   object DrainingControl {
 
     /**
-     * Combine control and a stream completion signal materialized values into
+     * Combine the consumer control and a stream completion signal materialized values into
      * one, so that the stream can be stopped in a controlled way without losing
      * commits.
+     *
+     * For use in `mapMaterializedValue`.
      */
     def apply[T](tuple: (Control, Future[T])) = new DrainingControl[T](tuple._1, tuple._2)
+
+    /**
+     * Combine the consumer control and a stream completion signal materialized values into
+     * one, so that the stream can be stopped in a controlled way without losing
+     * commits.
+     *
+     * For use in the `toMat` combination of materialized values.
+     */
+    def form[T]: (Control, Future[T]) => DrainingControl[T] = new DrainingControl[T](_, _)
```

Comment:
We could just overload the constructor for the scaladsl and the factory method for javadsl, instead of coming up with new names.  It seems to work locally.

Reply:
You're right, I tried different ways and the compiler was unhappy, but now it works.